File: frontend/types/order.ts
export type OrderStatus =
  | 'pending'
  | 'preparing'
  | 'out_for_delivery'
  | 'delivered'
  | 'cancelled'
  | 'skipped';

export interface DailyOrder {
  id: string;
  user_id: string;
  subscription_id: string;
  date: string;
  status: OrderStatus;
  received_status: boolean;
  delivery_address: string;
  area: string;
  created_at: string;
  // Joined fields for admin UI
  customer_name?: string;
  phone?: string;
  subscription_plan_name?: string;
  payment_status?: string;
}

File: backend/app/schemas/order.py
from pydantic import BaseModel
from typing import Optional

class UpdateOrderStatusRequest(BaseModel):
    status: str # pending | preparing | out_for_delivery | delivered | cancelled | skipped

class DailyOrderResponse(BaseModel):
    id: str
    user_id: str
    subscription_id: str
    date: str
    status: str
    received_status: bool
    delivery_address: str
    area: str
    created_at: Optional[str] = None
    customer_name: Optional[str] = None
    phone: Optional[str] = None
    subscription_plan_name: Optional[str] = None
    payment_status: Optional[str] = None
